include/service/entities/site_query_service.js

```diff
@@ -259,19 +259,68 @@ module.exports = function SiteQueryServiceModule(pb) {
    * @param {Function} cb
    */
   SiteQueryService.prototype.getCollections = function (cb) {
-      this.listCollections({}, function(err, items) {
-          if(pb.util.isError(err)) {
-            pb.log.error(err);
-            return cb(err);
-          }
+    this.listCollections({}, function(err, items) {
+      if(pb.util.isError(err)) {
+        pb.log.error(err);
+        return cb(err);
+      }
 
-          items = items.filter(function(item) {
-              return item.name.indexOf('system.indexes') === -1 && item.name.indexOf('system.namespaces') === -1;
-          });
+      items = items.filter(function(item) {
+        return item.name.indexOf('system.indexes') === -1 && item.name.indexOf('system.namespaces') === -1;
+      });
+
+      cb(err, items);
+    });
+  };
+
+  /**
+   * Deletes all site specific content
+   * @param {Array} collections -  array of collection names
+   * @param {String} siteid - unique site id
+   * @param {Function} callback - callback function
+   */
+  SiteQueryService.prototype.deleteSiteSpecificContent = function (collections, siteid, callback) {
+    var self = this;
+    var BaseObjectService = pb.BaseObjectService;
+
+    SiteService.siteExists(siteid, function (err, exists) {
+      if (util.isError(err)) {
+        return callback(err);
+      }
+
+      if (!exists) {
+        return callback(BaseObjectService.validationError([BaseObjectService.validationFailure("siteid", "Invalid siteid")]));
+      }
 
-          cb(err, items);
+      var tasks = util.getTasks(collections, function (collections, i) {
+        return function (taskCallback) {
+          self.delete({site: siteid}, collections[i].name, function (err, commandResult) {
+            if (util.isError(err) || !commandResult) {
+              return taskCallback(err);
+            }
+
+            var numberOfDeletedRecords = commandResult.result.n;
+            pb.log.silly(numberOfDeletedRecords + " site specific " + collections[i].name + " records associated with " + siteid + " were deleted");
```

Comment:
It is more performant to restructure this as:
```
pb.log.silly("%s site specific %s records associated with %s were deleted", numberOfDeletedRecords,  collections[i].name, siteid);
```
The reason is because in the event that the log level is not "silly" then it is cheaper to just put the parameters on the call stack than it is to do the string concatenation, pass it to the function, then find out the log level will not allow for logging the statement.